principles/evidence/software-development.md

```
@@ -0,0 +1,49 @@
+- We've seen the code do what we want several times
+- We can't produce any scenarios where the code doesn't do what we want
+
+Happy path thinking skips the last and very crucial step. The step where we
+actually create reproducible scenarios (aka "tests") and exhaust ourselves and
+our creativity in producing more.
+
+Common culprits include
+
+- Failing to test error/edge cases
+- Failing to test different families of numerics
+    - `0`, `1`, plurals, negatives, infinites, NaNs, floats, etc.
```

markdownlint-cli2 (0.17.2)

Unordered list indentation
Expected: 2; Actual: 4

(MD007, ul-indent)